src/poliastro/twobody/orbit.py

```diff
@@ -294,12 +297,17 @@ def sample(self, values=None, function=propagate):
 
         elif isinstance(values, int):
             if self.ecc < 1:
-                nu_values = np.linspace(0, 2 * np.pi, values) * u.rad
+                # first sample eccenric anomaly, then transform into true anomaly
+                E_values = np.linspace(0, 2 * np.pi, values) * u.rad
+                nu_values = E_to_nu(E_values, self.ecc)
+                
             else:
                 # Select a sensible limiting value for non-closed orbits
                 # This corresponds to r = 3p
                 nu_limit = np.arccos(-(1 - 1 / 3.) / self.ecc)
-                nu_values = np.linspace(-nu_limit, nu_limit, values)
+                E_limit = nu_to_E(nu_limit, self.ecc) / u.rad
```

Comment:
Not sure if the eccentric anomaly makes sense in the context of hyperbolic orbits. I wouldn't be surprised if this fails.